docs/content/debugging.md

```diff
@@ -0,0 +1,119 @@
+during policy evaluation as well as seeing how many times a particular line of code is executed.
+
+See the [print function documentation](../policy-reference/#debugging) for more details on how to use
+the `print` built-in function in different contexts.
+
+## Ecosystem Projects
+
+{{< ecosystem_feature_embed key="debugging-rego" topic="Debugging Rego" >}}
+
+# Debugging OPA Instances in Distributed Systems
+
+Debugging problems in distributed systems poses a number of challenges. Since OPA is commonly deployed in a distributed
+fashion, as part of a larger platform, it is helpful to understand the various tools and techniques available for
+debugging OPA in these environments.
+
+## OPA Logs
+
+OPA logs are a great place to start when debugging issues. The logs can be used to understand what OPA is doing
+at any given time. Common issues such as failing to load in policy or data bundles will be shown here.
+
+You can also enable debug logging to get more detailed information about what OPA is doing with `--log-level debug`.
+This is documented in the [CLI documentation](../cli/#options-10) for `opa run`.
+
+### Decision Logging
+
+When OPA responds to a query, it is making a decision based on the policy and data that it has loaded. With the default
+logging configuration, these are not logged in detail to the OPA logs. However, it is possible to enable console decision
+logging by setting the following in OPA's config file:
+
+```yaml
+decision_logs:
+  console: true
+```
+
+It might be preferable to send these logs to an HTTP endpoint or other system, to learn more about decision logging,
+take a look at the [Decision Logging documentation](../management-decision-logs).
+
+## Metrics, Heath and Status APIs
+
+Like other cloud-native tools, OPA exposes `/metrics` and `/health` endpoints that can be used to understand the
+state of an OPA instance at any given time.
+
+* `/metrics` - exposes Prometheus metrics about the OPA instance's memory use, bundle loading and HTTP requests.
+  Read more in the [Metrics documentation](../monitoring).
+* `/health` - shows information about the instance's readiness to serve requests, there are options available to also
+  show information about the loading of bundles and other plugins. Read more about the endpoint in the
+  [Health API documentation](../rest-api/#health-api).
+* `/status` - is a JSON formatted endpoint that shows both health and metrics information. Read more in
+  [Status API documentation](../rest-api/#status-api).
+
+## Manually Querying OPA
+
+In distributed systems, it's common that an OPA instance is being invoked by another service, sometimes it can be helpful
+to isolate the OPA instance and query it directly. This can be done using the [REST API](../rest-api).
+
+For example, to get a snapshot of the data that OPA has loaded, you can use the following command:
+
+```shell
+curl --silent http://$OPA_HOSTNAME/v1/data/
```

Comment:
If we assume $OPA_HOSTNAME other than localhost, should we perhaps use https in the URL here and below?

Either way, remove the trailing slash from `data/` as all you'll get from that is 
`<a href="/v1/data">Moved Permanently</a>.`

Reply:
Oops, good spot!